### PR TITLE
feat: implement blinky for nrf52833 on micro:bit

### PR DESCRIPTION
**Pull request by cyril-marpaud**
This example enables one to blink the first LED on [BBC micro:bit v2](https://en.wikipedia.org/wiki/Micro_Bit) (`nrf52833`-based board).

### CODE REVIEW

File: examples/nrf52833/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+[package]
+edition = "2021"
+name = "embassy-nrf52840-examples"
```
[the remark this answers is not in the document]
Comment (cyril-marpaud) [reply examples/nrf52833/Cargo.toml]: ✅

File: examples/nrf52833/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+embassy-net = { version = "0.2.0", path = "../../embassy-net", features = ["defmt", "tcp", "dhcpv4", "medium-ethernet"], optional = true }
+embassy-usb = { version = "0.1.0", path = "../../embassy-usb", features = ["defmt", "msos-descriptor",], optional = true }
+embedded-io = { version = "0.6.0", features = ["defmt-03"]  }
+embedded-io-async = { version = "0.6.0", optional = true, features = ["defmt-03"] }
+embassy-lora = { version = "0.1.0", path = "../../embassy-lora", features = ["time", "defmt"], optional = true }
+lora-phy = { version = "2", optional = true }
+lorawan-device = { version = "0.11.0", default-features = false, features = ["async", "external-lora-phy"], optional = true }
+lorawan = { version = "0.7.4", default-features = false, features = ["default-crypto"], optional = true }
+embassy-net-esp-hosted = { version = "0.1.0", path = "../../embassy-net-esp-hosted", features = ["defmt"], optional = true }
+embassy-net-enc28j60 = { version = "0.1.0", path = "../../embassy-net-enc28j60", features = ["defmt"], optional = true }
```
[the remark this answers is not in the document]
Comment (cyril-marpaud) [reply examples/nrf52833/Cargo.toml]: ✅

**Comment by cyril-marpaud** [comment]
> Thank you for the PR 🚀
> 
> Please also add the examples to `ci.sh`.
> 
> I'm not sure if this should live here vs in a board support crate for the microbit. On one hand, the microbit is a pretty widespread starter board, but there are others like the adafruit boards as well, and it's not clear to me where the line should be drawn and considered too much maintenance overhead (i.e. devkits from chip manufacturer is ok, but any non-chip manufacturer board is not?)
> 
> Would love to see more opinions on this before merging.

Totally agree about the fact that this example should live in a specific micro:bit folder but I couldn't find any in this repo (didn't look thoroughly though) so I went with `nrf52833` instead of `micro:bit` to follow the trend.
Given that no microcontroller is self-sufficient per se and requires a board to be exploitable (e.g. which LED does the nrf52840's blinky example blink ?), I also wonder how examples are usually added to each µC's folder.

**Comment by Dirbaio** [comment]
The examples are intended to showcase how to use the HAL. HAL usage is identical in all nrf52 chips, any nrf52840 example will work on the microbit if you change the chip in `Cargo.toml` and `.cargo/config.toml`.

I would love to have ready-to-go example for popular nrf52 boards, but I'm afraid the maintenance burden will be too high, similar to what we're already running into with the stm32 examples. (We have stm32 examples for each chip familiy because there is sometimes relevant hardware differences so the way you'd use the HAL changes. Still there's a lot of overlap, and when you change some HAL API you often have to go and update ~10 examples that are almost identical copy-pastes from each other)

Thanks anyway for the PR!

**Pull request closed without merging**